src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -1989,16 +1989,19 @@ jint os::init_2(void) {
     if (status != 0) {
       log_info(os)("os::init_2 getrlimit failed: %s", os::strerror(errno));
     } else {
+      // On macOS >= 10.6 if we define _DARWIN_UNLIMITED_STREAMS or _DARWIN_C_SOURCE
+      // (we define _DARWIN_C_SOURCE) we can ask for RLIM_INFINITY
       nbr_files.rlim_cur = nbr_files.rlim_max;
-
-#ifdef __APPLE__
-      // Darwin returns RLIM_INFINITY for rlim_max, but fails with EINVAL if
-      // you attempt to use RLIM_INFINITY. As per setrlimit(2), OPEN_MAX must
-      // be used instead
-      nbr_files.rlim_cur = MIN(OPEN_MAX, nbr_files.rlim_cur);
-#endif
-
       status = setrlimit(RLIMIT_NOFILE, &nbr_files);
+#ifdef __APPLE__
+      if (status != 0) {
```

Comment:
Should this be

`if (status != 0 && errno == EINVAL)`

so that any other error still gets logged?

Reply:
Yes, that would be a nice improvement. There are other places where this change would have to be implemented, i.e. os_aix, os_linux, but I think it's worth doing this right now, as opposed to filling a followup and going through another round of reviews.

Reply:
Good catch, done.